recipes/publish.rb

```diff
@@ -50,45 +50,17 @@
   #########################################################################
 
   # update a data bag with the artifact build info
-  # TODO: (jtimberman) This is not the first time this has been used in
-  # a delivery build cookbook. It's probably time to create a Chef
-  # resource for this functionality
-  ruby_block 'track-artifact-data' do # ~FC014
-    block do
-      load_delivery_chef_config
-      proj = Chef::DataBag.new
-      proj.name(project_name)
-      proj.save
+  load_delivery_chef_config
+  chef_data_bag project_name
 
-      proj_data = {
-        'id' => build_version,
-        'version' => build_version,
-        'artifact' => last_build_env.merge('type' => 'hart'),
-        'delivery_data' => node['delivery'],
-      }
-
-      proj_item = Chef::DataBagItem.new
-      proj_item.data_bag(proj.name)
-      proj_item.raw_data = proj_data
-      proj_item.save
-    end
+  chef_data_bag_item build_version do
+    raw_data('id' => build_version,
```

Comment:
I believe all of this data will need to be wrapped in `lazy` as the values you are sending in won't be available until the converge phase.